src/main/java/org/kiwiproject/dynamicproperties/PropertyExtractor.java

```diff
@@ -73,6 +83,15 @@ private static List<String> getListFromEnum(Class<?> enumClass) {
                 .collect(toList());
     }
 
+    private static List<Choice> getListFromChoiceSupplier(Class<? extends ChoiceSupplier> choiceSupplier) {
+        try {
+            return choiceSupplier.getDeclaredConstructor().newInstance().get();
+        } catch (InvocationTargetException | InstantiationException | IllegalAccessException | NoSuchMethodException e) {
+            LOG.error("exception getting choices from ChoiceSupplier", e);
```

Comment:
Maybe include the type here, so someone reading the log has an idea from reading the log message. e.g.:

```suggestion
            LOG.error("Error getting choices from ChoiceSupplier {}", choiceSupplier, e);
```